### PR TITLE
Remove client, data and cache

### PR DESCRIPTION
**Pull request by madadam**
This PR cherry-picks commits 660461cc11911759e216cc3555c15cfe4ffc7ab8 and 470235c3efdd031594a63cabc75ed404458ea44e from https://github.com/maidsafe/routing/pull/1797, and updates the tests so they still pass. 

It also removes `Request` and `Response` types - routing can now send arbitrary `Vec<u8>` as user message.

The `ci_test` and `key_value_store` examples have been commented out, but not removed yet. They should probably be moved over to vault in some form. 

The `tests/ci.rs` test has been commented out. A decision needs to be made whether it should be adapted to not use client or removed completely.

### CODE REVIEW

File: examples/ci_test.rs [hunk 1]
```diff
@@ -8,6 +8,8 @@
 
 //! Run a local network with 'nodes' nodes sending 'requests' requests.
 
+/* TODO: This example makes no sense without client, remove it
```
Comment (dirvine) [inline examples/ci_test.rs]: I would vote we do remove this in this PR actually. Saves time and clients (SCL) should do these kind of tests along with vaults themselves.

File: examples/key_value_store.rs [hunk 2]
```diff
@@ -6,6 +6,8 @@
 // KIND, either express or implied. Please review the Licences for the specific language governing
 // permissions and limitations relating to use of the SAFE Network Software.
 
+/* TODO: This example makes no sense without client, remove it
```
Comment (dirvine) [inline examples/key_value_store.rs]: Same as as above, this one is a vault test really. So we can remove from here.

File: examples/utils/example_client.rs [hunk 3]
```diff
@@ -7,10 +7,9 @@
 // permissions and limitations relating to use of the SAFE Network Software.
```
Comment (dirvine) [inline examples/utils/example_client.rs]: I think this whole file is removed in another commit, so happy we remove now or later as we grab those other commits.

File: tests/mock_network/messages.rs [hunk 4]
```diff
@@ -0,0 +1,116 @@
...
+                    content: ref req_content,
+                    ..
+                }) => {
+                    request_received_count += 1;
+                    if content == *req_content {
+                        break;
+                    }
+                }
+                Ok(_) => (),
+                _ => panic!("Event::RequestReceived not received"),
```
Comment (fizyk20) [inline tests/mock_network/messages.rs]: Should this say `MessageReceived`?

File: tests/mock_network/messages.rs [hunk 5]
```diff
@@ -0,0 +1,116 @@
...
+    let mut request_received_count = 0;
+
+    for node in nodes.iter_mut().filter(|n| n.is_recipient(&dst)) {
+        loop {
+            match node.try_next_ev() {
+                Ok(Event::MessageReceived { content, src, dst }) => {
+                    request_received_count += 1;
+                    if req_content == content {
+                        if let Err(err) = node.inner.send_message(dst, src, res_content.clone()) {
+                            trace!("Failed to send GetIData success response: {:?}", err);
```
Comment (fizyk20) [inline tests/mock_network/messages.rs]: Similar to above - there is no `GetIData` here, is there?

File: tests/mock_network/messages.rs [hunk 6]
```diff
@@ -0,0 +1,116 @@
...
+                        }
+                        break;
+                    }
+                }
+                Ok(_) => (),
+                _ => panic!("Event::RequestReceived not received"),
```
Comment (fizyk20) [inline tests/mock_network/messages.rs]: Same as above.

File: tests/mock_network/messages.rs [hunk 7]
```diff
@@ -0,0 +1,116 @@
...
+    let res_content: Vec<_> = rng.gen_iter().take(11).collect();
+
+    assert!(nodes[sender_index]
+        .inner
+        .send_message(src, dst, req_content.clone())
+        .is_ok());
+
+    let _ = poll_all(&mut nodes);
+
+    let mut request_received_count = 0;
```
Comment (fizyk20) [inline tests/mock_network/messages.rs]: Should the variable name be changed? (to `message_received_count` for example)

File: tests/mock_network/messages.rs [hunk 8]
```diff
@@ -0,0 +1,116 @@
...
+    loop {
+        match nodes[sender_index].inner.try_next_ev() {
+            Ok(Event::MessageReceived { content, .. }) => {
+                response_received_count += 1;
+                if res_content == content {
+                    break;
+                }
+            }
+            Ok(_) => (),
+            _ => panic!("Event::ResponseReceived not received"),
```
Comment (fizyk20) [inline tests/mock_network/messages.rs]: Same as above.

File: tests/mock_network/messages.rs [hunk 9]
```diff
@@ -0,0 +1,116 @@
...
+                _ => panic!("Event::RequestReceived not received"),
+            }
+        }
+    }
+
+    assert!(request_received_count >= quorum);
+
+    let _ = poll_all(&mut nodes);
+
+    let mut response_received_count = 0;
```
Comment (fizyk20) [inline tests/mock_network/messages.rs]: Same as above.

File: tests/mock_network/messages.rs (hunk 7 above)
Comment (madadam) [reply tests/mock_network/messages.rs]: I would keep the names here. Although we no longer have `Request` / `Response` as types, this test is still using them conceptually.

File: tests/mock_network/messages.rs (hunk 9 above)
Comment (madadam) [reply tests/mock_network/messages.rs]: See https://github.com/maidsafe/routing/pull/1801/files#r326533826

**Pull request merged**